library/std/src/sys/unix/fd.rs

```diff
@@ -28,7 +33,8 @@ const READ_LIMIT: usize = libc::ssize_t::MAX as usize;
 
 impl FileDesc {
     pub fn new(fd: c_int) -> FileDesc {
-        FileDesc { fd }
+        assert_ne!(fd, -1);
+        unsafe { FileDesc { fd } }
```

Comment:
```suggestion
        // SAFETY: we just asserted that the value is in the valid range and isn't `-1(0xFF_FF_FF_FE unsigned)`
        unsafe { FileDesc { fd } }
```

Reply:
@elichai `-1` as an unsigned value is all `F`s -- it does not end in an `E`.
```rust
-1i32 as u32 == 0xFFFF_FFFFu32

// what you wrote above is actually -2
-2i32 as u32 == 0xFFFF_FFFEu32
```

Reply:
Oopss Thanks!